concrete-core/src/backends/core/implementation/engines/ggsw_ciphertext_discarding_conversion.rs

```diff
@@ -0,0 +1,238 @@
+    ///
+    /// let mut fourier_ciphertext: GgswCiphertextComplex64 =
+    ///     engine.convert_ggsw_ciphertext(&ciphertext)?;
+    ///
+    /// // We're going to re-encrypt and re-convert the input with another secret key
+    /// // For this, it is required that the second secret key uses the same GLWE dimension
+    /// // and polynomial size as the first one.
+    /// let key_2: GlweSecretKey64 = engine.create_glwe_secret_key(glwe_dimension, polynomial_size)?;
+    ///
+    /// engine.discard_encrypt_ggsw_ciphertext(&key_2, &mut ciphertext, &plaintext, noise)?;
+    /// engine.discard_convert_ggsw_ciphertext(&mut fourier_ciphertext, &ciphertext)?;
+    ///
+    /// #
+    /// assert_eq!(fourier_ciphertext.glwe_dimension(), glwe_dimension);
+    /// assert_eq!(fourier_ciphertext.polynomial_size(), polynomial_size);
+    /// assert_eq!(fourier_ciphertext.decomposition_base_log(), base_log);
+    /// assert_eq!(fourier_ciphertext.decomposition_level_count(), level);
+    ///
+    /// engine.destroy(key_1)?;
+    /// engine.destroy(key_2)?;
+    /// engine.destroy(plaintext)?;
+    /// engine.destroy(ciphertext)?;
+    /// engine.destroy(fourier_ciphertext)?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn discard_convert_ggsw_ciphertext(
+        &mut self,
+        output: &mut GgswCiphertextComplex64,
+        input: &GgswCiphertext64,
+    ) -> Result<(), GgswCiphertextDiscardingConversionError<Self::EngineError>> {
+        GgswCiphertextDiscardingConversionError::perform_generic_checks(output, input)?;
+        unsafe { self.discard_convert_ggsw_ciphertext_unchecked(output, input) };
+        Ok(())
+    }
+
+    unsafe fn discard_convert_ggsw_ciphertext_unchecked(
+        &mut self,
+        output: &mut GgswCiphertextComplex64,
+        input: &GgswCiphertext64,
+    ) {
+        let input_bsk = StandardBootstrapKey::from_container(
+            input.0.as_tensor().as_container().clone(),
+            input.glwe_dimension().to_glwe_size(),
+            input.polynomial_size(),
+            input.decomposition_level_count(),
+            input.decomposition_base_log(),
+        );
+        let output_impl = ImplFourierBootstrapKey::allocate(
+            Complex64::new(0., 0.),
+            input.glwe_dimension().to_glwe_size(),
+            input.polynomial_size(),
+            input.decomposition_level_count(),
+            input.decomposition_base_log(),
+            LweDimension(1),
+        );
+        let mut output_bsk = FourierLweBootstrapKey64(output_impl);
+        let buffers = self.get_fourier_bootstrap_u64_buffer(&output_bsk);
+        output_bsk.0.fill_with_forward_fourier(&input_bsk, buffers);
+        output
+            .0
+            .as_mut_tensor()
+            .as_mut_slice()
+            .clone_from_slice(output_bsk.0.as_tensor().as_slice());
+    }
+}
+
+impl<Ciphertext> GgswCiphertextDiscardingConversionEngine<Ciphertext, Ciphertext> for CoreEngine
```

Comment:
Here I don't understand why we need this function (again).

Reply:
Answered above.